test/plugins/test-trace-google-gax.ts

```diff
@@ -13,52 +13,81 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-'use strict';
 
-var common = require('./common'/*.js*/);
-var assert = require('assert');
-var path = require('path');
+/**
+ * google-gax exports createApiCall, a transform on a function that accepts an
+ * argument and callback (representing arbitrary asynchronous work), as well as
+ * metadata and call options (details specific to gRPC). Internally, it does the
+ * necessary work to generate metadata and call options for each call to this
+ * function, as well as scheduling retries, etc. so that the caller doesn't
+ * have to be concerned with these details. The returned, transformed function
+ * has a different call signature as a result.
+ *
+ * There is a caveat to the above -- the input is not the pre-transformed
+ * function itself, but rather a *Promise* that will resolve to that function.
+ *
+ * Because every call of the transformed function must defer the actual
+ * asynchronous work until the above mentioned Promise has been resolved,
+ * createApiCall presents a possible source of context loss. The test in this
+ * file ensures that spans corresponding to that asynchronous work and spans
+ * created in its callback are created with the correct root span context, which
+ * is the context in which the transformed function is called, *not* the context
+ * in which createApiCall was called.
+ */
 
-describe('google-gax', function() {
-  var agent;
-  var speech;
+import * as assert from 'assert';
 
-  before(function() {
-    agent = require('../../..').start({
-      projectId: '0',
-      keyFilename: path.join(__dirname, '..', 'fixtures', 
-          'gcloud-credentials.json'),
-      enhancedDatabaseReporting: true,
-      samplingRate: 0
-    });
-    const SpeechClient = require('./fixtures/google-cloud-speech1').SpeechClient;
-    speech = new SpeechClient({
-      projectId: '0',
-      keyFilename: path.join(__dirname, '..', 'fixtures', 
-          'gcloud-credentials.json')
-    });
+import * as trace from '../trace';
+
+interface ApiCallSettings {
+  merge: () => {
+    otherArgs: {}
+  };
+}
+type Callback<T> = (err: Error|null, res?: T) => void;
+type InnerApiCall<I, O> =
+    (request: I, metadata: {}, options: {}, callback: Callback<O>) => void;
+type OuterApiCall<I, O> =
```

Comment:
\<rant\> I still don't understand the google-gax api... but that's not the problem of trace agent... \</rant\>

Reply:
🤣

Reply:
At its core it's a `Promise<Function> => Function` transform